lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,597 @@
+   - This module will sort resulting C(pg_hba) files if a rule change is required.
+     This could give unexpected results with manual created hba files, if it was improperly sorted.
+     For example a rule was created for a net first and for a ip in that net range next.
+     In that situation, the 'ip specific rule' will never hit, it is in the C(pg_hba) file obsolete.
+     After the C(pg_hba) file is rewritten by the M(pg_hba) module, the ip specific rule will be sorted above the range rule.
+     And then it will hit, which will give unexpected results.
+   - With the 'order' parameter you can control which field is used to sort first, next and last.
+   - The module supports a check mode and a diff mode.
+
+requirements:
+    - "ipaddress"
+
+author: Sebastiaan Mannem (github.com/sebasmannem)
+'''
+
+EXAMPLES = '''
+- name: Grant user joe from network 192.168.0.0/24 access to database sales using md5 password authentication.
+  pg_hba:
+    user=joe
+    source=192.168.0.100/24
+    database=sales
+    method=md5
+
+- name: Grant user repmgr from host 192.168.0.100/32 access for replication without password.
+  pg_hba:
+    user=repmgr
+    source=192.168.0.100
+    database=replication
+    method=trust
+'''
+
+RETURN = '''
+msgs:
+    description: List of textual messages what was done
+    returned: always
+    type: list
+    sample:
+       "msgs": [
+          "Removing",
+          "Changed",
+          "Writing"
+        ]
+backup_file:
+    description: File that the original pg_hba file was backed up to
+    returned: changed
+    type: string
+    sample: /tmp/pg_hba_jxobj_p
+pg_hba:
+    description: List of the pg_hba rules as they are configured in the specified hba file
+    returned: always
+    type: list
+    sample:
+      "pg_hba": [
+         {
+            "db": "all",
+            "method": "md5",
+            "src": "samehost",
+            "type": "host",
+            "usr": "all"
+         }
+      ]
+'''
+
+import os
+import re
+try:
+    import ipaddress
+    HAS_IPADDRESS = True
+except ImportError:
+    HAS_IPADDRESS = False
+import copy
+from ansible.module_utils.basic import AnsibleModule, tempfile, shutil
+
+PG_HBA_METHODS = ["trust", "reject", "md5", "password", "gss", "sspi", "krb5", "ident", "peer",
+                  "ldap", "radius", "cert", "pam"]
+PG_HBA_TYPES = ["local", "host", "hostssl", "hostnossl"]
+PG_HBA_ORDERS = ["sdu", "sud", "dsu", "dus", "usd", "uds"]
+PG_HBA_HDR = ['type', 'db', 'usr', 'src', 'mask', 'method', 'options']
+
+WHITESPACES_RE = re.compile(r'\s+')
+
+
+class PgHbaError(Exception):
+    '''
+    This exception is raised when parsing the pg_hba file ends in an error.
+    '''
+    pass
+
+
+class PgHba(object):
+    """
+    PgHba object to read/write entries to/from.
+    pg_hba_file - the pg_hba file almost always /etc/pg_hba
+    """
+    def __init__(self, pg_hba_file=None, order="sdu", backup=False, create=False):
```

Comment:
value of keyword parameter `order` is already `sdu` because `sdu` is the default value for `order` module parameter. Same for `backup` and `create`.

Reply:
Ok, no need to change.